src/Markup/Avalonia.Markup.Xaml/PortableXaml/AvaloniaRuntimeTypeProvider.cs

```diff
@@ -38,6 +38,7 @@ public class AvaloniaRuntimeTypeProvider : IRuntimeTypeProvider
             typeof(DataTemplate).GetTypeInfo().Assembly,
             typeof(SolidColorBrush).GetTypeInfo().Assembly,
             typeof(Binding).GetTypeInfo().Assembly,
+            typeof(DataGrid).GetTypeInfo().Assembly
```

Comment:
Again, this means that all avalonia applications will load the `DataGrid` assembly.